### PR TITLE
Annotation structure

### PR DESCRIPTION
**Pull request by linusnorton**
Basic annotation data structures

### CODE REVIEW

File: projects/media-viewer/src/lib/media-viewer/annotations/annotation-api.service.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    const url = `/em-anno/annotations/${annotation.id}`;
+
+    return this.httpClient.delete<Annotation>(url, { observe: 'response' });
+  }
+
+  public createAnnotation(annotation: Annotation): Observable<HttpResponse<Annotation>> {
+    const url = `/em-anno/annotations`;
+
+    return this.httpClient.post<Annotation>(url, annotation, { observe: 'response' });
+  }
```
Comment (velimd) [inline projects/media-viewer/src/lib/media-viewer/annotations/annotation-api.service.ts]: All fine, just a questions, would we also need edit as well as if a comment changed or the annotations position when its dragged/resized

File: projects/media-viewer/src/lib/media-viewer/annotations/annotations.module.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  ],
+  entryComponents: [
+  ],
+  providers: [
+    AnnotationApiService
+  ],
+  exports: [
+  ]
+})
+export class AnnotationsModule { }
```
Comment (velimd) [inline projects/media-viewer/src/lib/media-viewer/annotations/annotations.module.ts]: i did this however, as my changes arent here anymore, maybe you want to add this module in the media-viewer module?

File: projects/media-viewer/src/lib/media-viewer/annotations/annotation-api.service.ts (hunk 1 above)
Comment (linusnorton) [reply projects/media-viewer/src/lib/media-viewer/annotations/annotation-api.service.ts]: I think we will. I did not see it in the document-viewer api code though

File: projects/media-viewer/src/lib/media-viewer/annotations/annotations.module.ts (hunk 2 above)
Comment (linusnorton) [reply projects/media-viewer/src/lib/media-viewer/annotations/annotations.module.ts]: ok

**Pull request merged**